observability/metrics/src/main/scala/com/daml/metrics/ExecutorServiceMetrics.scala

```diff
@@ -162,4 +192,55 @@ object ExecutorServiceMetrics {
 
   }
 
+  /*
+   * Handles the removal of registered gauges for the executor service.
```

Comment:
This class may be used specifically for cleaning up gauges, however, I'd move this comment to where we instantiate this class and leave this class' name generic (see suggestion below) and only describe the side-effect it has on shutdown here. This should avoid confusion around the actual implementation not referring to anything related to metrics.

Reply:
Yes that was my initial idea as well, but for it to fully work then it would also have to be moved outside the `metrics` module and package. Ideally it would be in the `executors` module but that would create a circular dependency, we would need a new module to get around that by I decided that the extra effort just for that is not really worth it. 
If you think the rename would provide some value I'm more than happy to do it.

Reply:
Makes sense. I'm OK with keeping this class here and making it package-private if we only expect it to be used from`com.daml.metrics` (though still would suggest renaming it and slightly rephrasing the comment to clarify the only use-case we have for this class).